src/Data/Auth/AuthProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { AuthConfig, LocalConfig } from "./PrimaryAuthController";
+import DatabaseAuthProvider from "./Providers/DatabaseAuthProvider";
+import KeycloakAuthContext from "./Providers/KeycloakAuthContext";
+import NoAuthProvider from "./Providers/NoAuthProvider";
+
+interface AuthProviderProps {
+  children: React.ReactNode;
+  config: undefined | AuthConfig | LocalConfig;
+  keycloakUrl: string | undefined;
+}
+
+const AuthProvider = ({ children, config, keycloakUrl }: AuthProviderProps) => {
```

Comment:
Why can't this just be a function with typed arguments?

Reply:
Correct me if I understood question wrong, but as we auth providers return context component as a wrapper we have to follow the React's convention for deconstructed arguments